pkg/arvo/sys/hoon.hoon

```diff
@@ -1641,25 +1637,23 @@
         b
       ?:  (mor p.n.a p.n.b)
         ?:  =(p.n.b p.n.a)
-          [n.b $(a l.a, b l.b) $(a r.a, b r.b)]
+          :+  [p.n.a (meg p.n.a q.n.a q.n.b)]
+            $(b l.b, a l.a)
+          $(b r.b, a r.a)
         ?:  (gor p.n.b p.n.a)
-          $(a [n.a $(a l.a, b [n.b l.b ~]) r.a], b r.b)
-        $(a [n.a l.a $(a r.a, b [n.b ~ r.b])], b l.b)
-      ?:  =(p.n.a p.n.b)
-        :+  [p.n.a (meg p.n.a q.n.a q.n.b)]
-          $(b l.b, a l.a)
-        $(b r.b, a r.a)
+          $(a a(l $(a l.a, b b(r ~))), b r.b)
+        $(a a(r $(a r.a, b b(l ~))), b l.b)
       ?:  (gor p.n.a p.n.b)
-        $(b [n.b $(b l.b, a [n.a l.a ~]) r.b], a r.a)
-      $(b [n.b l.b $(b r.b, a [n.a ~ r.a])], a l.a)
+        $(b b(l $(b l.b, a a(r ~))), a r.a)
+      $(b b(r $(b r.b, a a(l ~))), a l.a)
```

Comment:
I'm replacing all instances of formats like `b b(...`, in this case it gets reduced to this: `$(r.b $(b r.b, l.a ~), a l.a)`